src/Symfony/Component/Yaml/Tests/ParserTest.php

```diff
@@ -2362,6 +2362,39 @@ public function testParseValueWithNegativeModifiers()
             $this->parser->parse($yaml)
         );
     }
+
+    /**
+     * This is a regression test for a bug where a YAML block with a nested multiline string using | was parsed without
+     * a trailing \n when a shorter YAML document was parsed before.
+     *
+     * When a shorter document was parsed before, the nested string did not have a \n at the end of the string, because
+     * the Parser thought it was the end of the file, even though it is not.
+     */
+    public function testParsingMultipleDocuments()
+    {
+        $shortDocument = 'foo: bar';
+        $longDocument = <<<YAML
+a:
+    b: |
+        row
+        row2
+c: d
+YAML;
+
+        $expected = ['a' => ['b' => "row\nrow2\n"], 'c' => 'd'];
+
+        // The parser was not used before, so there is a new line after row2
+        $this->assertSame($expected, $this->parser->parse($longDocument));
+
+        $parser = new Parser();
```

Comment:
as far as I can see we can simply use `$this->parser` when parsing the two documents

Reply:
see #38235